source/dub/generators/cmake.d

```diff
@@ -0,0 +1,132 @@
+/**
+    Generator for CMake build scripts
+
+    Copyright: © 2015 Steven Dwy
+    License: Subject to the terms of the MIT license, as written in the included LICENSE.txt file.
+    Authors: Steven Dwy
+*/
+module dub.generators.cmake;
+
+import dub.compilers.buildsettings;
+import dub.generators.generator;
+import dub.internal.vibecompat.core.log;
+import dub.internal.vibecompat.core.file;
+import dub.project;
+
+import std.algorithm: map, uniq, sort;
+import std.array: appender, join;
+import std.stdio: File, write;
+import std.string: format, replace;
+
+class CMakeGenerator: ProjectGenerator
+{
+    this(Project project)
+    {
+        super(project);
+    }
+    
+    override void generateTargets(GeneratorSettings settings, in TargetInfo[string] targets)
+    {
+        auto script = appender!(char[]);
+        auto scripts = appender!(string[]);
+        bool[string] visited;
+        string projectRoot = m_project.rootPackage.path.toString;
+        string cmakeListsPath = "%s/CMakeLists.txt".format(projectRoot);
```

Comment:
Would be better to use Path to concat paths, try `m_project.rootPackage.path ~ "CMakeLists.txt"` instead.
There is [`Path.toNativeString()`](http://vibed.org/api/vibe.inet.path/Path) to get the result.